ocs_ci/ocs/monitoring.py

```diff
@@ -299,3 +299,145 @@ def check_if_monitoring_stack_exists():
     else:
         logger.info("Monitoring stack is not configured on the cluster")
         return False
+
+
+def get_prometheus_response(api, query) -> dict:
+    """
+    Get the response from Prometheus based on the provided query
+
+    Args:
+        api (PrometheusAPI): A PrometheusAPI object
+        query (str): The Prometheus query string
+
+    Returns:
+        dict: A dictionary representing the parsed JSON response from Prometheus
+    """
+    resp = api.get("query", payload={"query": query})
+    if resp.ok:
+        logger.debug(query)
+        logger.debug(resp.text)
+        return json.loads(resp.text)
+
+
+def get_pvc_namespace_metrics():
+    """
+    Get PVC and Namespace metrics from Prometheus.
+
+    Returns:
+        dict: A dictionary containing the PVC and Namespace metrics data
+    """
+
+    api = ocs_ci.utility.prometheus.PrometheusAPI()
+
+    pvc_namespace = {}
+
+    logger.info("Get PVC namespace data from Prometheus")
+
+    # use get_prometheus_response to store the response text to a dict
+    pvc_namespace["PVC_NAMESPACES_BY_USED"] = get_prometheus_response(
+        api,
+        "sum by (namespace, persistentvolumeclaim) "
+        "(kubelet_volume_stats_used_bytes{namespace='${namespace}'} * "
+        "on (namespace, persistentvolumeclaim) "
+        "group_left(storageclass, provisioner) (kube_persistentvolumeclaim_info * on (storageclass) "
+        "group_left(provisioner) kube_storageclass_info "
+        "{provisioner=~'(.*rbd.csi.ceph.com)|(.*cephfs.csi.ceph.com)|(ceph.rook.io/block)'}))",
+    )
+    pvc_namespace["PVC_NAMESPACES_TOTAL_USED"] = get_prometheus_response(
+        api,
+        "sum(sum by (namespace, persistentvolumeclaim) "
```

Comment:
the inner sum query seems to be equivalent to the one used for "PVC_NAMESPACES_BY_USED". Suggest creating a constant for it

Reply:
done